package/MDAnalysis/lib/_augment.pyx

```diff
@@ -0,0 +1,307 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+#
+
+import cython
+import numpy
+cimport numpy
+
+from libcpp.vector cimport vector
+from libc.math cimport sqrt
+
+__all__ = ['augment', 'undo_augment']
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def augment(float[:, ::1] coordinates, float[:, ::1] dm, float r):
+    """Calculate augmented coordinate set
+

```

Comment:
Add: "Note, does not return coordinates from the initial array, to merge these use `np.concatenate(a, b)`" (or is it `np.concat`?)

Reply:
Do you think it should be added within the function itself since undo_augment already assumes that  results are after the merge? Or will there be any benefit in keeping it in the same way? One could, in principle, provide 0 as nreal, but I am not sure how will that be useful?